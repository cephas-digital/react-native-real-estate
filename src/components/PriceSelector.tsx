import React, { PureComponent } from "react";
import { Input } from "react-native-elements";
import { ListingFilters } from "../interfaces";

interface MyProps {
  setFilters(filters: ListingFilters): void;
  filters: ListingFilters;
}

export default class PriceSelector extends PureComponent<MyProps> {
  getPriceAsString() {
    if (
      this.props.filters.maxPrice === null ||
      typeof this.props.filters.maxPrice === "undefined"
    ) {
      return "";
    }

    return this.props.filters.maxPrice.toString();
  }
  render() {
    return (
      <Input
        containerStyle={{ width: 160 }}
        value={this.getPriceAsString()}
        placeholder="Max Price"
        leftIcon={{ name: "dollar", type: "font-awesome" }}
        leftIconContainerStyle={{ marginRight: 10 }}
        onChangeText={(maxPrice: string) => {
          if (isNaN(parseInt(maxPrice, 10))) {
            this.props.setFilters({ maxPrice: null });
          } else {
            this.props.setFilters({ maxPrice: parseInt(maxPrice, 10) });
          }
        }}
        keyboardType={"number-pad"}
      />
    );
  }
}
