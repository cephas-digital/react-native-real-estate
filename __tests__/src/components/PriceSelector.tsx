import React from "react";
import renderer from "react-test-renderer";
import PriceSelector from "../../../src/components/PriceSelector";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <PriceSelector
        filters={{}}
        setFilters={filters => console.log(filters)}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
