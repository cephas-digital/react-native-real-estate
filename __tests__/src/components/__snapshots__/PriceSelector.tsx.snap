// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 4,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#86939e",
        "flexDirection": "row",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Max Price"
      style={
        Object {
          "alignSelf": "center",
          "color": "black",
          "flex": 1,
          "fontSize": 16,
          "minHeight": 40,
        }
      }
      testID="RNE__Input__text-input"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
